packages/bpk-component-breakpoint/src/useMediaQuery.ts

```diff
@@ -16,25 +16,24 @@
  * limitations under the License.
  */
 
-import { useState, useEffect } from 'react';
+import { useEffect, useState } from 'react';
+
+const useMediaQuery = (query: string, matchSSR = false): boolean => {
+  const isClient = typeof window !== 'undefined' && !!window.matchMedia;
 
-const useMediaQuery = (query: string): boolean => {
   const [matches, setMatches] = useState(
-    window.matchMedia ? window.matchMedia(query).matches : false,
+    isClient ? window.matchMedia(query).matches : matchSSR,
```

Comment:
For my own benefit, what does "matches" mean in this context? Particularly in the context of `matchSSR`?

The docs mention:

> You can pass matchSSR which will instruct the breakpoint to match any time it is rendered on the server.

But this isn't clear to me. Maybe it's worth improving our guidance around this as I think our guidance is pretty light here

Reply:
> Does it mean if matchSSR is true this component will always render on the server?

Yes, exactly

> Do we attempt to guess the screen size on the server at all?

Nope, this isn't something this component has done, which is why it has the "what do you want to match on the server" approach.

> But this isn't clear to me. Maybe it's worth improving our guidance around this as I think our guidance is pretty light here

I can have a think about re-phrasing this, feel free to suggest something that is clearer to someone that doesn't have the internal workings in their head

Reply:
I was just having a play around and in Skyscanner via request-context we have access to the booleans `isMobile`, `isTablet` and `isBrowser`(?). 

My first option would be to change the API to `alwaysRenderWhenSSR`, as this is unambiguous what will happen and the word `match` on it's own isn't clear in this context. If we don't want to change the API, expanding our docs would be the next best option. 

Something like the below with an explanation would help convey that any consumer of this API would want to make at least an informed guess on the device size for SSR, otherwise if you have multiple `BpkBreakpioint` components, chances are the majority of users will see a flash of one component in the wrong device setting, following by the correct component once the component has mounted and executed the client side only code

```jsx
// `matchSSR` - Always render this component on the server. You will likely want to make an informed guess from the request object what size the users device is.
<BpkBreakpoint query={BREAKPOINTS.TABLET} matchSSR={isMobilePhone}>
  {mobileContent}
<BplBrealpoint/>
```

Reply:
Yup I think that's an improvement in the guidance, I like the push towards using other data to estimate what breakpoint is likely to match on the client-side. Pushed a change that is similar to this, let me know what you think.